src/core/utility/datatypes/byte_array.cpp

```diff
@@ -489,7 +489,7 @@ namespace argos {
    CByteArray& CByteArray::operator>>(float& f_value) {
       double fDoubleValue;
       *this >> fDoubleValue;
-      f_value = fDoubleValue;
+      f_value = static_cast<float>(fDoubleValue);
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Yes--you get a double -> float implicit conversion warning without it